### DISCUSSION TITLE
Ollama models are lost after every Watchtower upgrade

### DISCUSSION THREAD

**Discussion by sendmebits** [open-webui/open-webui#7264] [General]
I use Open WebUI with bundled Ollama support for CPU. 

Everytime I use the recommended Watchertower commend to upgrade, I lose all Ollama models. Open WebUI upgrades fine and keeps all the settings. 

`docker run --rm --volume /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --run-once open-webui`

Is there an option to maintain all models/settings on upgrades?

**Comment by DaymondArmstrong** [comment]
Are you running your Docker with the -v option within your docker run command?  -v ollama:/root/.ollama -v open-webui:/app/backend/data as an example.

**Reply by DaymondArmstrong** [reply]
While I haven't configured Watchtower yet, and am still a Docker fledgling, I believe you still have to have the container running the =v command, which runs Ollama in a separate persistent docker volume, and open-webui in yet another separate persistent docker container. So, Watchtower should update without losing the models you have trained, as Watchtower should only update the open-webui volume.

I run my docker container using the following format:

docker run -d -p (External TCP Port):8080 -p (If you want additional access to Ollama: External TCP Port):11434 --gpus=all -v ollama:/root/.ollama -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:ollama

Hope that helps

**Reply by sendmebits** [reply]
I'm new to Docker as well! Long time Linux and Proxmox user.

I used the below to install and it does have -v both for Ollama and Open Web UI:

`docker run -d -p 3000:8080 -v ollama:/root/.ollama -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:ollama`

However, if I do a docker ps I get this, which makes it seem to me as though Ollama is running in the same container as Open WebUI:

`# docker ps
CONTAINER ID   IMAGE                                  COMMAND           CREATED      STATUS                PORTS                    NAMES
b651754f03f4   ghcr.io/open-webui/open-webui:ollama   "bash start.sh"   4 days ago   Up 4 days (healthy)   0.0.0.0:3000->8080/tcp   open-webui`

**Reply by sendmebits** [reply]
To add - Ollama is running and operation after using Watchtower to upgrade, however, it loses all its data/models...

**Reply by DaymondArmstrong** [reply]
I have been doing manual updates, so I cannot be certain of Watchtower and whether or not it should update just the volume that contains open-webui. In reading up on Watchtower, I was expecting it would work that way, but it would destroy a few projects for me, if I lost my models, as I have trained them on my data already. 

However, it isn't that difficult to manually update, as you just stop the docker container, remove the docker container, docker pull the latest open-webui, then run your startup command again. This is the process I have been using to update, and have not lost a single model since I did that.

I have had 4 full re-installations of Linux and started over with my models, and at least 2 of those was due to my inexperience with docker itself, the other 2 were me attempting to get Ubuntu 24 to run properly, and each time I upgraded from 22.04, it blew up docker and the container would not run.

I ran the following commands to update to the latest version of open-webui:

sudo docker stop open-webui
sudo docker rm open-webui
sudo docker pull ghcr.io/open-webui/open-webui:ollama

Then I ran this command to start my container:

docker run -d -p 3000:8080 -p 11434:11434 --gpus=all -v ollama:/root/.ollama -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:ollama

Keep in mind that I am running with a Nvidia GPU, hence the --gpus=all in my run command. I am going to experiment with Watchtower and will post an update here. Nice to learn something new with others.

**Comment by sendmebits** [comment]
Also, a 'docker volume ls' does give me 2 volumes:

```
DRIVER    VOLUME NAME
local     ollama
local     open-webui
```

The ollama models do appear here: 
> /var/lib/docker/volumes/ollama/_data/models

So the models are still there after the upgrade, it's just that Ollama no longer sees the volume.

**Reply by DaymondArmstrong** [reply]
That has to be something with Watchtower that neither of us understands at this moment. I have some time coming up and will finish taking a deep dive on Watchtower. I can at least vouch for the fact that the volumes work with the manual process, and I do not lose any data.

**Reply by sendmebits** [reply]
I investigated for a while, but I couldn't figure it out.

I ended up just installed Ollama separately instead of with the Open WebUI Docker container and I have it all setup now. Looks like it will be a mystery!
